src/core/absorption_chiller.jl

```diff
@@ -28,61 +28,217 @@
+    om_cost_per_kw::Float64 = NaN
+end
+
+function AbsorptionChiller(d::Dict;
+        chp_prime_mover::Union{String, Nothing} = nothing,
+        existing_boiler::Union{ExistingBoiler, Nothing} = nothing
+    )
+
+    # convert Vector{Any} from JSON dictionary to Vector{Float64}
+    for (k, v) in d
+        if typeof(v) <: AbstractVector{Any}
+            d[k] = convert(Vector{Float64}, v)  # JSON.parsefile makes things Vector{Any}
+        end
+    end
+
+    absorp_chl = AbsorptionChiller(; dictkeys_tosymbols(d)...)
+
+    #check for 0.0 max size, return nothing if so
+    if absorp_chl.max_ton == 0.0
+        @warn "0.0 kW provided as capacity for AbsoprtionChiller, this technology will be excluded."
+        return nothing
+    end
+
+    custom_ac_inputs = Dict{Symbol, Any}(
+        :hot_water_or_steam => absorp_chl.hot_water_or_steam,
+        :installed_cost_per_ton => absorp_chl.installed_cost_per_ton,
+        :cop_thermal => absorp_chl.cop_thermal,
+        :om_cost_per_ton => absorp_chl.om_cost_per_ton
+    )
+
+    htf_defaults_response = get_absorption_chiller_defaults(;max_ton=absorp_chl.max_ton,
+        hot_water_or_steam=absorp_chl.hot_water_or_steam, 
+        chp_prime_mover=chp_prime_mover, 
+        existing_boiler=existing_boiler
+    )
+    
+    #convert defaults for any properties not enetered
+    defaults = htf_defaults_response["default_inputs"]
+    for (k, v) in custom_ac_inputs
+        if isnothing(v)
+            setproperty!(absorp_chl, k, defaults[string(k)])
+        end
+    end
+
+    # generate derived inputs for use in JuMP model
+    absorp_chl.min_kw = absorp_chl.min_ton * KWH_THERMAL_PER_TONHOUR
+    absorp_chl.max_kw = absorp_chl.max_ton * KWH_THERMAL_PER_TONHOUR
+    absorp_chl.installed_cost_per_kw = absorp_chl.installed_cost_per_ton / KWH_THERMAL_PER_TONHOUR
+    absorp_chl.om_cost_per_kw = absorp_chl.om_cost_per_ton / KWH_THERMAL_PER_TONHOUR
+
+    return absorp_chl
+end
+
+
+"""
+get_absorption_chiller_defaults(prime_mover::String, boiler_type::String, size_class::Int)
 
-        min_kw = min_ton * KWH_THERMAL_PER_TONHOUR
-        max_kw = max_ton * KWH_THERMAL_PER_TONHOUR
-        installed_cost_per_kw = installed_cost_per_ton / KWH_THERMAL_PER_TONHOUR
-        om_cost_per_kw = om_cost_per_ton / KWH_THERMAL_PER_TONHOUR
-
-        new(
-            min_ton,
-            max_ton,
-            cop_thermal,
-            cop_electric,
-            installed_cost_per_ton,
-            om_cost_per_ton,
-            macrs_option_years,
-            macrs_bonus_fraction,
-            min_kw,
-            max_kw,
-            installed_cost_per_kw,
-            om_cost_per_kw
+return a Dict{String, Any} by selecting the appropriate values from 
+data/chp/absorption_chiller_defaults.json, which contains values based on heat transfer medium (hot_water_or_steam)
+such as:
+- "installed_cost_per_ton"
+- "om_cost_per_ton"
+- "cop_thermal"
+- "tech_sizes_for_cost_curve"
```

Comment:
Are we modeling a cost curve (y-intercept + slope) for AbsorptionChiller, or just linearly interpolating between the cost data points between two adjacent sizes? If not, I suggest we rename to something like tech_sizes_for_cost_**data**, since it's different than the CHP tech_sizes_for_cost_curve.

Reply:
It's the latter; I implemented the change in e2da3a96bf9a619170852a07942f6087bcc215df